### PR TITLE
Web Apps: Use SelectWoo, a more accessible fork of Select2, for comboboxes

### PR DESCRIPTION
**Pull request by solleks**
## Summary
https://dimagi-dev.atlassian.net/browse/USH-498

Two JAWS users report that the [Select2](https://github.com/select2/select2) comboboxes in Web Apps are not read by their screen reader. Specifically, they cannot hear any items in the dropdown list, although they can enter text in the edit text box.

The [SelectWoo](https://developer.woocommerce.com/2017/08/08/selectwoo-an-accessible-replacement-for-select2/) fork has accessibility improvements and it appears to solve the issues with JAWS, while maintaining feature compatibility with select2.

Since entrycontrols code is used in app preview, that is also switched to SelectWoo.

## Product Description

This change should only affect the behavior when a screen reader is used. It should cause the screen reader to announce the items in the dropdown listbox and it should be possible to navigate through the items with up/down arrows and select one with the Enter key.

### QA Plan

[QA ticket](https://dimagi-dev.atlassian.net/browse/QA-2371)

We will attempt to evaluate this change before release with one of the contact tracers who reported the issue.

### Rollback instructions

- [ X] This PR can be reverted after deploy with no further considerations

### CODE REVIEW

**Comment by solleks** [comment]
Hi @stephherbers. Here is a change to make JAWS read comboboxes and hopefully fix one of the issues reported by our NJ contact tracer.

File: corehq/apps/hqwebapp/templates/hqwebapp/base.html [hunk 1]
```diff
@@ -436,7 +436,7 @@
 
     {% if not requirejs_main %}
       {% compress js %}
-        <script src="{% static 'select2/dist/js/select2.full.min.js' %}"></script>
+        <script src="{% static 'select-woo/dist/js/selectWoo.full.min.js' %}"></script>
```
Comment (orangejenny) [inline corehq/apps/hqwebapp/templates/hqwebapp/base.html:439]: This file is the base template for most of HQ, you want `cloudcare/formplayer_home.html` to just update web apps.

**Review by orangejenny — COMMENTED** [review]
https://github.com/dimagi/commcare-hq/pull/28955 adds another select2 to web apps and will likely be merged before this, so once that's merged, this PR should be updated.

Web apps js tests include select2, in `cloudcare/spec/form_entry/mocha.html`, so that likely needs to be updated.

Tests are failing with a yarn error. I think you need to commit `yarn.lock` to fix them.
```
yarn install v1.22.10
[1/5] Validating package.json...
[2/5] Resolving packages...
error Your lockfile needs to be updated, but yarn was run with `--frozen-lockfile`.
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
```

File: corehq/apps/hqwebapp/templates/hqwebapp/base.html (hunk 1 above)
Comment (solleks) [reply corehq/apps/hqwebapp/templates/hqwebapp/base.html]: OK, I moved the import into `cloudcare/formplayer_home.html`.

**Comment by solleks** [comment]
I added a modified yarn.lock file and imported select-woo into `cloudcare/spec/form_entry/mocha.html`.

I merged with the master branch including PR #28955 and found that SelectWoo comboboxes were used in place of Select2 without requiring any more changes. I think this is because the select-woo code makes the JQuery selector '.select2' refer to itself if it is not already defined.

**Comment by orangejenny** [comment]
> I think this is because the select-woo code makes the JQuery selector '.select2' refer to itself if it is not already defined.

Huh, interesting. Was this in web apps or app preview? I thought web apps now now included both select2 (from base.html) and selectWoo (from formplayer_home.html).

**Comment by solleks** [comment]
> > I think this is because the select-woo code makes the JQuery selector '.select2' refer to itself if it is not already defined.
> 
> Huh, interesting. Was this in web apps or app preview? I thought web apps now now included both select2 (from base.html) and selectWoo (from formplayer_home.html).

This was in web apps. It looks like the selectWoo code is included before the select2 code.

**Comment by solleks** [comment]
I changed the selector value from .select2 to .selectWoo on the theory that it reflects our intent better.

**Comment by orangejenny** [comment]
Thanks, I agree it's clearer now.

**Review by orangejenny — APPROVED** [review]

**Comment by solleks** [comment]
Hi @stephherbers. Would you mind pushing this change to staging? Ben suggested we do that and he will look at it when time allows.

**Comment by solleks** [comment]
QA testing is done and our screen reader expert user also approved of this change, so I think we are good to merge this one.

**Pull request merged**